tmt/beakerlib.py

```diff
@@ -154,23 +182,27 @@ def fetch(self):
                     f"using ref '{library.ref}'.")
             self.parent.debug(f"Library '{self}' already fetched.", level=3)
             # Reuse the existing metadata tree
-            self.tree = library.tree
+            self.tree: fmf.Tree = library.tree
         # Fetch the library and add it to the index
         except KeyError:
             self.parent.debug(f"Fetch library '{self}'.", level=3)
             # Prepare path, clone the repository, checkout ref
+            assert self.parent.workdir
```

Comment:
What's the point of this assert?

Reply:
@psss  Because `self.parent.workdir` property defines the return type `Optional[str]`, this is required here to make `mypy` happy on line 190, where `join` does not take a None value. 

The other solution would be:
* to check it for `None` here and raise an exception
* fix the `workdir` property that it cannot return `None`

So?

Reply:
Can `self.parent.workdir` be `None` at this point, is it allowed, or does code flow absolutely rules such an option out? mypy cannot tell, `workdir` can be `None` and `None` cannot be passed to `join()`, but if humans *know* for sure `workdir` will no longer be none when tmt already works with libraries, then `assert` "narrows" the type for mypy.

An exception is a valid tool as well, changing `workdir` to never allow `None` might be complicated, I recall seeing some `workdir=None` parameters around in `tmt.utils.Common`.

Reply:
yeah, I would leave it to the expert :) let's wait

Reply:
@thrix, @happz, thanks for the detailed clarification. So here, the `workdir` will always be defined as we get `parent` from a real `discover.fmf` plugin instance. Also, if the `parent` is not provided, `Common(workdir=True)` is used which also ensures that a workdir is initialized. So, I guess, the `assert` way seems to be the most natural approach here.